Accounts.Tests/AccountServiceTest.cs

```diff
@@ -1,49 +1,66 @@
-using Accounts.Domain.Entities;
-using Accounts.Infrastructure.Persistence;
+﻿using Accounts.Core.Entities;
+using Accounts.Core.Models;
+using Accounts.Core.Ports.Driven;
+using Accounts.Core.Ports.Driving;
+using Accounts.Core.Services;
+using Accounts.Core.Validators;
 using FluentValidation;
+using FluentValidation.Results;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.EntityFrameworkCore.Diagnostics;
 using Moq;
 
 namespace Accounts.Tests
 {
-    public class AccountServiceTest : IDisposable
+    public class AccountServiceTest
     {
-        private readonly AppDbContext _dbContext;
+        private readonly Mock<IUserRepository> _userRepositoryMock;
+        private readonly Mock<IAddressRepository> _addressRepositoryMock;
+        private readonly Mock<ICityRepository> _cityRepositoryMock;
+        private readonly Mock<IContactInfoRepository> _contactInfoRepositoryMock;
+        private readonly Mock<ILoginInfoRepository> _loginInfoRepositoryMock;
+        private readonly Mock<ITransactionHandler> _transactionHandlerMock;
+        //private readonly Mock<IValidator<RegisterUserCommand>> _validatorMock;
+        private readonly RegisterUserValidator _validator;
+        private readonly Mock<IPasswordHasher> _passwordHasherMock;
+        private readonly Mock<ISessionStore> _sessionStoreMock;
+
         private readonly AccountService _service;
-        private readonly Mock<IValidator<RegisterUserRequest>> _validatorMock;
-        
+
         public AccountServiceTest()
         {
-            var options = new DbContextOptionsBuilder<AppDbContext>()
-                .UseInMemoryDatabase(databaseName: "TestDb")
-                .ConfigureWarnings(warnings => warnings.Ignore(InMemoryEventId.TransactionIgnoredWarning))
-                .Options;
-
-
-            _dbContext = new AppDbContext(options);
-
-            
-            // Mock the validator
-            _validatorMock = new Mock<IValidator<RegisterUserRequest>>();
-            _validatorMock.Setup(v => v.ValidateAsync(It.IsAny<RegisterUserRequest>(), default))
-                .ReturnsAsync(new FluentValidation.Results.ValidationResult());
+            _userRepositoryMock = new Mock<IUserRepository>();
+            _addressRepositoryMock = new Mock<IAddressRepository>();
+            _cityRepositoryMock = new Mock<ICityRepository>();
+            _contactInfoRepositoryMock = new Mock<IContactInfoRepository>();
+            _loginInfoRepositoryMock = new Mock<ILoginInfoRepository>();
+            _transactionHandlerMock = new Mock<ITransactionHandler>();
+            //_validatorMock = new Mock<IValidator<RegisterUserCommand>>();
+            _validator = new RegisterUserValidator();
+            _passwordHasherMock = new Mock<IPasswordHasher>();
+            _sessionStoreMock = new Mock<ISessionStore>();
+
+            //_validatorMock.Setup(v => v.ValidateAsync(It.IsAny<RegisterUserCommand>(), default))
+            //    .ReturnsAsync(new ValidationResult());
```

Comment:
udkommenterede mocks og deres kald / brug kunne også fjernes hvis ikke anvendes eller ikke er vigtige for core funktionalitet